### ISSUE TITLE
Option uri_adapter not working with ruby 2.1

### ISSUE THREAD

**Issue by rtsinani** [jnunemaker/httparty#468]
Trying to use:

`uri_adapter Addressable::URI`

on ruby 2.1, raises the same error as if we were using URI. It looks like the error is in `Net::HTTP.proxy_uri` which calls `URI` again [here](https://github.com/ruby/ruby/blob/ruby_2_1/lib/net/http.rb#L1031).

Is there a fix for this, or am I looking at the wrong place?

Thanks

**Comment by jnunemaker** [comment]
I don't think there is much httparty can do about that. I'm closing this for now, but if you have any ideas, I'd be curious.

**Comment by rtsinani** [comment]
We fixed it by patching `Net::HTTP` in ruby 2.1 with the code from latest ruby:

```ruby
module Net
  class HTTP
    # The proxy URI determined from the environment for this connection.
    def proxy_uri # :nodoc:
      @proxy_uri ||= URI::HTTP.new(
        "http".freeze, nil, address, port, nil, nil, nil, nil, nil
      ).find_proxy
    end
  end
end
```

But I thought you might have a better idea. Cheers for taking the time.
